src/utils/text.cc

```diff
@@ -171,14 +170,7 @@ void mbs_write_wrapped(ostream & out, const string & text,
     bytes_read = mbrtowc (&wc, s, s_bytes, &shift_state);
     if (bytes_read > 0)
```

Comment:
note (unrelated to patch), that the special -2 and -1 cases are not handled here - and this is an unsigned int, so it will still go through the if part.